src/ccache.cpp

```diff
@@ -965,6 +965,12 @@ to_cache(Context& ctx,
   Util::unsetenv("DEPENDENCIES_OUTPUT");
   Util::unsetenv("SUNPRO_DEPENDENCIES");
 
+#ifndef __APPLE__
+  // Older Clang versions on macOS didn't support this.
+  if (ctx.config.is_compiler_group_clang()) {
+    args.push_back("--");
```

Comment:
Why always include `--`? Would it be possible to only include it when it's on the original command line?